File: easy/2490_circular_sentence/solution.cpp
using namespace std;

#include <string>

class Solution {
public:
  bool isCircularSentence(string sentence) {
    int spacePos = sentence.find(' ');

    while (spacePos != -1) {
      if (sentence[spacePos - 1] != sentence[spacePos + 1])
        return false;

      spacePos = sentence.find(' ', spacePos + 1);
    }

    return sentence[0] == sentence[sentence.size() - 1];
  }
};

File: easy/2490_circular_sentence/solution.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        space_pos = sentence.find(" ")

        while space_pos != -1:
            if sentence[space_pos - 1] != sentence[space_pos + 1]:
                return False

            space_pos = sentence.find(" ", space_pos + 1)

        return sentence[0] == sentence[-1]
